src/snowflake/connector/result_batch.py

```diff
@@ -258,6 +258,16 @@ def uncompressed_size(self) -> Optional[int]:
     def column_names(self) -> List[str]:
         return [col.name for col in self.schema]
 
+    @property
```

Comment:
We should not make this a `property`. We shouldn't `urlsplit` for every log line we do. Let's calculate this in `__init__` and then reuse that.